code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -978,6 +987,36 @@
 	set category = "Robot Commands"
 	set_autosay()
 
+/mob/living/silicon/robot/proc/control_headlamp()
+	if(stat)
+		src << "<span class='danger'>This function is currently offline.</span>"
+		return
+
+	if(lamp_intensity < lamp_max) //Lamp is set lower than the max.
+		lamp_intensity += 2
+		src << "Headlamp power set to Level [lamp_intensity/2]."
+	else //Greater than or equal to the max, so turn it off instead.

```

Comment:
you can use modulo for this, as this is clearly clock arithmetic.

lamp_intensity = lamp_intensity % your_max + 1
This works like this:
1 = 1 % 5 +1 = 2
2 = 2 % 5 +1 = 3
3 = 3 % 5 +1 = 4
4 = 4 % 5 + 1 = 5
5 = 5 % 5 + 1 = 1 (X Mod X is always 0)

this is a MUCH nicer way to format a "cycle" between 1 and a Max value.